core/controllers/tasks_test.py

```diff
@@ -21,6 +21,7 @@
 from core.domain import feedback_services
 from core.domain import rights_domain
 from core.domain import suggestion_services
+from core.domain import taskqueue_services
```

Comment:
It's a bit strange to me that we're replacing the task queue in this way. Could we instead keep all tests & production code using the existing task queue prod & test APIs, but have it internally point to the new task queue service? That would significantly reduce the size of the PR & help it be focused.

Reply:
@BenHenning -- sorry, I'm not sure I understand your suggestion here. We do a switchover at the platform layer for dev/prod but stuff above that layer is generic. The various implementations of the platform layer implement the same API. Is this consistent with what you are thinking?

Reply:
To rephrase: I would expect that the task queue API defined in the platform layer would stay the same if the functionality is the same, and that wouldn't require changing everything; instead, we'd just be changing the implementation of the platform layer. Or, was task queue not sufficiently abstracted to allow that?

Reply:
Oh, I see. Yeah, it's less straightforward than that, because the API changes in Python 3. E.g. defer() no longer exists.

Reply:
Yes, I agree with sean, we are moving defer to our own implementation which is why I moved it out.